dataset/src/main/java/org/keycloak/benchmark/dataset/DatasetResourceProvider.java

```diff
@@ -457,6 +462,17 @@ private void addUserCreationTasks(RealmContext context, Task task, DatasetConfig
         }
     }
 
+    private List<PasswordCredentialModel> initializeCredentials(DatasetConfig config, RealmContext context) {
+        PasswordPolicy policy = context.getRealm().getPasswordPolicy();
+        PasswordHashProvider hash = policy.getHashAlgorithm() != null ?
+              baseSession.getProvider(PasswordHashProvider.class, policy.getHashAlgorithm()) :
+              baseSession.getProvider(PasswordHashProvider.class);
+
+        return IntStream.range(0, config.getUniqueCredentialCount())
```

Comment:
Probably should compute the minimum between the number of users and the number of credentials. 
```suggestion
        return IntStream.range(0, Math.min(config.getUsersPerRealm(), config.getUniqueCredentialCount()))
```